dart_application_2/bin/validations_util.dart

```diff
@@ -0,0 +1,166 @@
+import 'exceptions.dart';
+import 'package:validators/validators.dart';
+
+class ValidationsUtil {
+  /**
+   * given number should be numeric
+   */
+  static bool _isNumeric(String data) {
+    bool exceptionFlag = true;
+    try {
+      double value = double.parse(data);
+    } catch (e) {
+      exceptionFlag = false;
+    }
+    return exceptionFlag;
+  }
+
+  /**
+   * given option should be natural number
+   */
+  static bool _nonNatural(String data) {
+    bool exceptionFlag = false;
+    try {
+      int value = int.parse(data);
+      if (value <= 0) throw NonNaturalException();
+    } on NonNaturalException {
+      exceptionFlag = true;
+    } catch (e) {
+      exceptionFlag = true;
+    }
+    return exceptionFlag;
+  }
+
+  static bool _nonWhole(String data) {
+    bool exceptionFlag = false;
+    try {
+      int value = int.parse(data);
+      if (value < 0) throw NonWholeException();
+    } on NonWholeException {
+      exceptionFlag = true;
+    } catch (e) {
+      exceptionFlag = true;
+    }
+    return exceptionFlag;
+  }
+
+  /**
+   * given option is out of range
+   */
+  static bool _inRange(String option, int rangeStart, int rangeEnd) {
+    if (int.parse(option) >= rangeStart && int.parse(option) <= rangeEnd)
+      return true;
+    else
+      return false;
+  }
+
+  // given string should contain all alphabets
+  static bool _isAlphabet(String name) {
+    if (isAlpha(name)) return true;
+    return false;
+  }
+  /**
+   *this method checks if given name starts with capital letter
+   */
+
+  static bool _isNotCapital(String name) {
+    if (name[0].toUpperCase() == name[0]) return false;
+    return true;
+  }
+
+  /**
+   * validate given option:
+   * criteria:
+   * 1. Not empty
+   * 2. Should be numeric
+   * 3. Natural Number
+   * 4. Within given range
+   */
+  static bool validateOption(String option, int rangeStart, int rangeEnd) {
+    bool exceptionFlag = false;
+    try {
+      if (option.isEmpty) throw new EmptyStringException();
+      if (!_isNumeric(option)) throw new NonNumericException();
+      if (_nonWhole(option)) throw new NonWholeException();
+      if (!_inRange(option, rangeStart, rangeEnd))
```

Comment:
Pass int.parse(option) here cause you've already validated for it being a number above 
So by the time it reaches here it will be a number for sure